src/main/java/com/ausregistry/jtoolkit2/se/DomainAddRem.java

```diff
@@ -47,16 +49,48 @@ public DomainAddRem(AddRemType type, String[] nameservers,
         }
     }
 
+    /**
+     * Maximal specification of the attribute values which may be added or
+     * removed from a domain. Each of the parameters is optional, but at least
+     * one must be specified.
+     */
+    public DomainAddRem(AddRemType type, Host[] nameservers,
+        String[] techContacts, String[] adminContacts,
+        String[] billingContacts, Status[] statuses) {
+        this.type = type.toString();
+
+        if (nameservers != null) {
+            this.hosts = nameservers.clone();
+        }
+        if (techContacts != null) {
+            this.techContacts = techContacts.clone();
+        }
+        if (adminContacts != null) {
+            this.adminContacts = adminContacts.clone();
+        }
+        if (billingContacts != null) {
+            this.billingContacts = billingContacts.clone();
+        }
+        if (statuses != null) {
+            this.statuses = statuses.clone();
+        }
```

Comment:
This new constructor duplicates a lot of code from the original one. It can be simplified by calling `this(..)`.

Reply:
I have handled this please check